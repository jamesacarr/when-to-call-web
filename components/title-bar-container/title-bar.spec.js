import React from 'react';
import { shallow } from 'enzyme';
import TitleBar from './title-bar';

const classes = {
  root: 'root',
  wrapper: 'wrapper',
  search: 'search',
  input: 'input',
};

describe('<TitleBar/>', () => {
  describe('.render', () => {
    const performQuery = () => {};
    const setVisible = () => {};

    it('renders correctly when not loading', () => {
      const wrapper = shallow(<TitleBar classes={classes} performQuery={performQuery} setVisible={setVisible} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly when loading', () => {
      const wrapper = shallow(
        <TitleBar classes={classes} performQuery={performQuery} setVisible={setVisible} loading />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('.handleChange', () => {
    let wrapper;
    let performQuery;
    let setVisible;

    beforeEach(() => {
      performQuery = jest.fn();
      setVisible = jest.fn();

      wrapper = shallow(<TitleBar classes={classes} performQuery={performQuery} setVisible={setVisible} />);
      wrapper.instance().handleChange({ target: { value: 'abc' } });
    });

    it('sets query', () => {
      expect(wrapper.state('query')).toEqual('abc');
    });

    it('calls performQuery', () => {
      expect(performQuery).toHaveBeenCalledWith('abc');
    });
  });
});
