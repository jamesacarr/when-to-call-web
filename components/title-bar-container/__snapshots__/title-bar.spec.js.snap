// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBar/> .handleChange updates the value of the input 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <div
      className="wrapper"
    >
      <div
        className="search"
      >
        <pure(Search) />
      </div>
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search for business"
        value="abc"
      />
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;

exports[`<TitleBar/> .render renders correctly when loading 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <div
      className="wrapper"
    >
      <div
        className="search"
      >
        <pure(Search) />
      </div>
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search for business"
        value=""
      />
      <div>
        <WithStyles(CircularProgress)
          color="inherit"
          size={25}
        />
      </div>
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;

exports[`<TitleBar/> .render renders correctly when not loading 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <div
      className="wrapper"
    >
      <div
        className="search"
      >
        <pure(Search) />
      </div>
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search for business"
        value=""
      />
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
